src/lib/core/datetime.c

```diff
@@ -284,9 +287,7 @@ datetime_parse_full(struct datetime *date, const char *str, size_t len,
 	str += n;
 
 exit:
-	date->epoch =
-		((int64_t)dt_rdn(dt) - DT_EPOCH_1970_OFFSET) * SECS_PER_DAY +
-		sec_of_day - offset * 60;
+	date->epoch = dt_epoch(dt) + sec_of_day - offset * 60;
```

Comment:
Note - probably base epoch should be UTC only, without offset applied. Need to check Olson rules